test/functional/tests/io_class/test_io_class_pinning_eviction.py

```diff
@@ -0,0 +1,111 @@
+
+from api.cas import ioclass_config, casadm
+
+cache_size = Size(256, Unit.MB)
+core_size = Size(2, Unit.GiB)
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_pinning_eviction():
+    """
+    title: IO class pinning eviction - stress.
+    description: |
+        Check Open CAS ability to prevent from eviction of pinned IoClass - stress test.
+    pass_criteria:
+        - IoClasses loaded successfully
+        - Pinned class occupies the same ammount of cache after each IO operation on different IoClass
+    """
+
+    with TestRun.step("Prepare devices"):
+        cache, core = prepare(core_size=core_size, cache_size=cache_size)
+
+    with TestRun.step("Mount filesystem"):
+        core.create_filesystem(Filesystem.xfs)
+        core.mount(mountpoint)
+
+    with TestRun.step("Prepare test dirs"):
+        IoclassConfig = namedtuple("IoclassConfig", "id eviction_prio max_occupancy dir_path")
+        io_classes = [
+            IoclassConfig(1, 0, 0.60, f"{mountpoint}/A"),
+            IoclassConfig(2, 10, 0.70, f"{mountpoint}/B"),
+            IoclassConfig(3, 40, 0.80, f"{mountpoint}/C"),
+            IoclassConfig(4, 50, 0.90, f"{mountpoint}/D"),
+            IoclassConfig(5, 100, 0.70, f"{mountpoint}/E"),
+        ]
+
+        for io_class in io_classes:
+            fs_utils.create_directory(io_class.dir_path, parents=True)
+
+    with TestRun.step("Remove old config"):
+        ioclass_config.remove_ioclass_config()
+        ioclass_config.create_ioclass_config(False)
+
+    with TestRun.step("Add default ioclasses"):
+        ioclass_config.add_ioclass(*str(IoClass.default(allocation="0.00")).split(","))
+
+    with TestRun.step("Add IoClasses for dirs"):
+        for io_class in io_classes:
+            ioclass_config.add_ioclass(
+                io_class.id,
+                f"directory:{io_class.dir_path}&done",
+                io_class.eviction_prio,
+                f"{io_class.max_occupancy:0.2f}",
+            )
+        pinned_io_class = io_classes[0]
+        casadm.load_io_classes(cache_id=cache.cache_id, file=default_config_file_path)
+
+    with TestRun.step("Reset cache stats"):
+        cache.purge_cache()
+        cache.reset_counters()
+
+    with TestRun.step("Check occupancy "):
+        for io_class in io_classes:
+            occupancy = get_io_class_occupancy(cache, io_class.id)
+            if occupancy.get_value() != 0:
+                TestRun.fail(
+                    f"Incorrect inital occupancy for ioclass id: {io_class.id}."
+                    f" Expected 0, got: {occupancy}"
+                )
+
+    with TestRun.step(f"Trigger IO to pinned class directory"):
+        run_io_dir(
+            f"{pinned_io_class.dir_path}/tmp_file",
+            int((pinned_io_class.max_occupancy * cache_size) / Unit.Blocks4096),
+        )
+        pinned_occupancy = get_io_class_occupancy(cache, 1)
```

Comment:
I suggest using `pinned_io_class.id` instead of the raw number. The same for line 107